### PR TITLE
ECA Tests Verification - Adding tests to module

### PR DESCRIPTION
**Pull request by gromeroar**
## Description

Additional tests and code fixes were added to Membership Services. 
- In ECA module, the ReadCertificatePair function has been fixed, which indirectly fixes the TestReadCertificatePairBadIdentity test that throws an error when receives an invalid identity argument. 
- The following tests were added to eca_tests: 
  - TestReadCertificateByHash 
  - TestReadCertificateByInvalidHash.
## Motivation and Context

Growing the test coverage of the ECA module and fixing some code bugs that were discovered during the tests generation.
## How Has This Been Tested?

Running ECA tests with “go test” and running the unit tests using “make unit-test”.  
## Checklist:

- [X] I have added a [Signed-off-by](https://github.com/hyperledger/fabric/blob/master/CONTRIBUTING.md#legal-stuff).
- [X] This change requires no new documentation.
- [X] I have either added unit tests to cover my changes.
- [X] I have run [golint](https://github.com/golang/lint) and have fixed valid warnings in code I have added or modified. This tool generates false positives so you may choose to ignore some warnings. The goal is clean, consistent, and readable code.

The continuous integration build process will run [make checks](https://github.com/hyperledger/fabric/blob/master/Makefile#L22) to confirm that tests pass and that code quality meets minimum standards. You may optionally run this locally as PRs will not be accepted until they pass.

Signed-off-by:
Guillermo Romero : gromeroar guillermo.romero@gmail.com / rologui@ar.ibm.com

### CODE REVIEW

File: membersrvc/ca/eca.go [hunk 1]
```diff
@@ -410,16 +410,21 @@ func (ecap *ECAP) ReadCertificatePair(ctx context.Context, in *pb.ECertReadReq)
...
+			hasResults = true
 			var raw []byte
 			err = rows.Scan(&raw)
 			certs = append(certs, raw)
 		}
 		err = rows.Err()
 	}
 
+	if !hasResults {
+		return nil, errors.New("No certificates for the given Identity were found.")
```
Comment (srderson) [inline membersrvc/ca/eca.go]: Why is "Identity" capitalized in this error message?

**Comment by gromeroar** [comment]
Hi @srderson I've fixed this typo mistakes and found other ones so I've fixed too.

**Comment by srderson** [comment]
Thanks, but it looks lie there's a merge conflict now. Could you please rebase on master?

**Comment by gromeroar** [comment]
@srderson I finished with the task.

**Comment by srderson** [comment]
Just noticed that Travis failed. Looks unrelated, but going to rerun

**Comment by srderson** [comment]
Passing now. LGTM

**Pull request merged**